src/app/diary-module/components/diary-card/diary-card.component.ts

```diff
@@ -0,0 +1,39 @@
+import { Component, Input, OnInit } from '@angular/core'
+import { FormControl } from '@angular/forms'
+import { Card } from '../../util/card'
+
+@Component({
+    selector: 'app-diary-card',
+    templateUrl: './diary-card.component.html',
+    styleUrls: ['./diary-card.component.scss'],
+})
+export class DiaryCardComponent implements OnInit {
+    @Input()
+    card!: Card
+
+    text = new FormControl('')
+    visibility = new FormControl('')
+    btnName = new FormControl('SHOW MORE')
+
```

Comment:
you don't have to use any of these controls here. you cloud achieve this using `template literals` and `getter` with one state variable.

Reply:
the way you implemented this is not correct..